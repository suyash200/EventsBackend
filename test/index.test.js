import request from 'supertest'

let token
const url = 'http://localhost:4000/'






test('Post /auth/login', async () => {
    const res = await request(url).post('auth/login').send({ "email": "root@rot.com", "password": "root" })
    expect(res.status).toBe(200)
    expect(res.body).toHaveProperty('token')
    token = res.body.token
})


test('Post /event/getevent', async () => {
    const res = await request(url).get('events/getevents').auth(token, { type: "bearer" })
    console.log(res.body)
    expect(res.status).toBe(200)
    expect(res.body).toBeDefined()
    res.body.forEach(element => {
        expect(element).toMatchSnapshot({
            _id: expect.any(String)
        })
    });
})
