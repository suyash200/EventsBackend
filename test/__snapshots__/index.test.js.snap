// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /event/getevent  To get events 1`] = `
{
  "__v": 0,
  "_id": Any<String>,
  "bookedtickets": 0,
  "categories": [
    "technology",
  ],
  "createdAt": "2023-06-19T07:55:10.249Z",
  "createdBy": "6490009fe6bda83adb21306d",
  "date": "2020-12-03T18:30:00.000Z",
  "description": "an intresting events for fellow students",
  "eventName": "Vit Event",
  "guestSpeakers": [
    {
      "SpeakerName": "ram pandey",
      "_id": "649009de31a533dfa76be3f6",
      "description": "backend systems",
      "topic": [
        "backend systems",
      ],
    },
    {
      "SpeakerName": "Deepankar Bhade",
      "_id": "649009de31a533dfa76be3f7",
      "description": "Golang systems",
      "topic": [
        "Golang systems",
      ],
    },
  ],
  "hosts": [
    "some extrovert kid",
  ],
  "mode": "online",
  "paid": false,
  "peoplelimit": 400,
  "updatedAt": "2023-06-19T07:55:10.249Z",
  "venue": "audi",
}
`;

exports[`GEt /event/getevent 1`] = `
{
  "__v": 0,
  "_id": Any<String>,
  "bookedtickets": 0,
  "categories": [
    "technology",
  ],
  "createdAt": "2023-06-19T07:55:10.249Z",
  "createdBy": "6490009fe6bda83adb21306d",
  "date": "2020-12-03T18:30:00.000Z",
  "description": "an intresting events for fellow students",
  "eventName": "Vit Event",
  "guestSpeakers": [
    {
      "SpeakerName": "ram pandey",
      "_id": "649009de31a533dfa76be3f6",
      "description": "backend systems",
      "topic": [
        "backend systems",
      ],
    },
    {
      "SpeakerName": "Deepankar Bhade",
      "_id": "649009de31a533dfa76be3f7",
      "description": "Golang systems",
      "topic": [
        "Golang systems",
      ],
    },
  ],
  "hosts": [
    "some extrovert kid",
  ],
  "mode": "online",
  "paid": false,
  "peoplelimit": 400,
  "updatedAt": "2023-06-19T07:55:10.249Z",
  "venue": "audi",
}
`;

exports[`Post /event/getevent 1`] = `
{
  "__v": 0,
  "_id": Any<String>,
  "bookedtickets": 0,
  "categories": [
    "technology",
  ],
  "createdAt": "2023-06-19T07:55:10.249Z",
  "createdBy": "6490009fe6bda83adb21306d",
  "date": "2020-12-03T18:30:00.000Z",
  "description": "an intresting events for fellow students",
  "eventName": "Vit Event",
  "guestSpeakers": [
    {
      "SpeakerName": "ram pandey",
      "_id": "649009de31a533dfa76be3f6",
      "description": "backend systems",
      "topic": [
        "backend systems",
      ],
    },
    {
      "SpeakerName": "Deepankar Bhade",
      "_id": "649009de31a533dfa76be3f7",
      "description": "Golang systems",
      "topic": [
        "Golang systems",
      ],
    },
  ],
  "hosts": [
    "some extrovert kid",
  ],
  "mode": "online",
  "paid": false,
  "peoplelimit": 400,
  "updatedAt": "2023-06-19T07:55:10.249Z",
  "venue": "audi",
}
`;
